File: EASendMail/Samples_NetDesktop/VB/GmailOauth/AppProject1/FormOauth.vb
Public Class FormOauth
    Public Property OauthWrapper As OauthDesktopWrapper = Nothing

    Public Sub New()
        MyBase.New()
        InitializeComponent()
        Me.DialogResult = DialogResult.Cancel
    End Sub

    Private Sub FormOauth_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
        OauthBrowser.Navigate(OauthWrapper.OauthProvider.GenerateFullAuthUri())
    End Sub

    Private Sub OauthBrowser_DocumentCompleted(ByVal sender As Object, ByVal e As WebBrowserDocumentCompletedEventArgs) Handles OauthBrowser.DocumentCompleted
        Dim elment As HtmlElement = OauthBrowser.Document.GetElementById("code")
        If elment Is Nothing Then
            Return
        End If

        Dim code As String = elment.GetAttribute("value")
        OauthWrapper.AuthorizationCode = code
        Me.DialogResult = DialogResult.OK
        Me.Close()
    End Sub
End Class

File: EASendMail/Samples_NetDesktop/C#/GmailOauth/AppProject1/FormOauth.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using EASendMail.Oauth;

namespace AppProject1
{
    public partial class FormOauth : Form
    {
        public OauthDesktopWrapper OauthWrapper = null;
        public FormOauth()
        {
            InitializeComponent();
            this.DialogResult = DialogResult.Cancel;
        }

        private void FormOauth_Load(object sender, EventArgs e)
        {
            //start to open Gmail OAUTH login web page.
            OauthBrowser.Navigate(OauthWrapper.OauthProvider.GenerateFullAuthUri()); 
        }

        private void OauthBrowser_DocumentCompleted(object sender, WebBrowserDocumentCompletedEventArgs e)
        {
            HtmlElement elment = OauthBrowser.Document.GetElementById("code");
            if (elment == null)
            {
                return;
            }

            string code = elment.GetAttribute("value");
            OauthWrapper.AuthorizationCode = code;
            this.DialogResult = DialogResult.OK;
            this.Close();
        }
    }
}
